### PR TITLE
refs #3173: add appsv1beta1 and appsv1beta2 apiVersion 

### PR DESCRIPTION
**Pull request by paolomainardi**
Related to #3173

### CODE REVIEW

**Comment by unguiculus** [comment]
Could you also add `apps/v1/ReplicaSet`, `apps/v1beta2/DaemonSet`, and `apps/v1/DaemonSet`?

**Comment by paolomainardi** [comment]
@unguiculus done

File: pkg/kube/wait.go [hunk 1]
```diff
@@ -88,6 +88,36 @@ func (c *Client) waitForResources(timeout time.Duration, created Result) error {
 					currentDeployment,
 				}
 				deployments = append(deployments, newDeployment)
+			case *appsv1beta1.Deployment:
+				currentDeployment, err := kcs.ExtensionsV1beta1().Deployments(value.Namespace).Get(value.Name, metav1.GetOptions{})
```
Comment (unguiculus) [inline pkg/kube/wait.go]: Looking at this, I wonder if it shouldn't use the client corresponding to the API version.

`kcs.AppsV1beta1().Deployments`

If this is a problem, then it has already existed wit the old code.

Comment (mattfarina) [reply pkg/kube/wait.go]: @unguiculus I thought the same thing and tried to test this change. The problem is that `deploymentutil.GetNewReplicaSet` (a package from within k8s), which is used shortly after this and is passed the `currentDeployment`, takes as an input something using the extensions v1beta1 interface. Making your suggested switch causes that code to break.

I expect there is a better way to clean this up but I'm not sure it needs to happen in this PR.

**Comment by mattfarina** [comment]
The code makes sense and I've tested a few variations. It appears to be working.

/lgtm

**Review by mattfarina — APPROVED** [review]

**Comment by kkehres** [comment]
is there any way this can support openshift too?

kind: DeploymentConfig
apiVersion: apps.openshift.io/v1

apiVersion: autoscaling/v1
kind: HorizontalPodAutoscaler

**Pull request merged**